src/common/string/format_number.ts

```diff
@@ -9,7 +9,7 @@ import { FrontendTranslationData, NumberFormat } from "../../data/translation";
  */
 export const formatNumber = (
   num: string | number,
-  locale?: FrontendTranslationData,
+  locale?: Partial<FrontendLocaleData>,
```

Comment:
Why `Partial`?

Reply:
My thought was: The number formatting has no relation to the time formatting, but if it is not `Partial` we always need to specify a time format as well. In the normal UI code that is no issue, since `hass.locale` is the complete interface, but it initially felt weird to have to pass a `time_format` in the UI tests. But changing application code for the sake of tests, is a bad sign.

=> Will remove the `Partial`.